crates/polars-plan/src/dsl/function_expr/index_of.rs

```diff
@@ -28,9 +28,11 @@ pub(super) fn index_of(s: &mut [Column]) -> PolarsResult<Column> {
         // If the Series is sorted, we can use an optimized binary search to
         // find the value.
         IsSorted::Ascending | IsSorted::Descending
-            if !needle.is_null() &&
-            // search_sorted() doesn't support decimals at the moment.
-            !series.dtype().is_decimal() =>
+            if !(needle.is_null()
+                 // search_sorted() doesn't support these types at the moment:
```

Comment:
I think this should be supported. For decimals we should extract the `Int128` in both the haystack and the needle. 

And for the nested types we should convert both to the row-encoding.

Reply:
I can get that to that in a couple of weeks, hopefully. Could still merge this issue as a short-term workaround for `index_of()`, or just close it.

Reply:
I have added list and array support to `search_sorted`, to the extent possible. Decimals apparently work? So not search why that was disabled in `index_of`.